DataFormats/L1TParticleFlow/interface/PFCluster.h

```diff
@@ -0,0 +1,76 @@
+#ifndef DataFormats_L1TParticleFlow_PFCluster_h
+#define DataFormats_L1TParticleFlow_PFCluster_h
+
+#include <vector>
+#include "DataFormats/L1Trigger/interface/L1Candidate.h"
+#include "DataFormats/Common/interface/Ref.h"
+
+namespace l1t {
+
+  class PFCluster : public L1Candidate {
+  public:
+    /// constituent information. note that this is not going to be available in the hardware!
+    typedef std::pair<edm::Ptr<l1t::L1Candidate>, float> ConstituentAndFraction;
+    typedef std::vector<ConstituentAndFraction> ConstituentsAndFractions;
+
+    PFCluster() {}
+    PFCluster(float pt,
+              float eta,
+              float phi,
+              float hOverE = 0,
+              bool isEM = false,
+              float ptError = 0,
+              int hwpt = 0,
+              int hweta = 0,
+              int hwphi = 0)
+        : L1Candidate(PolarLorentzVector(pt, eta, phi, 0), hwpt, hweta, hwphi, /*hwQuality=*/isEM ? 1 : 0),
+          hOverE_(hOverE),
+          ptError_(ptError) {
+      setPdgId(isEM ? 22 : 130);  // photon : non-photon(K0)
+    }
+    PFCluster(
+        const LorentzVector& p4, float hOverE, bool isEM, float ptError = 0, int hwpt = 0, int hweta = 0, int hwphi = 0)
+        : L1Candidate(p4, hwpt, hweta, hwphi, /*hwQuality=*/isEM ? 1 : 0), hOverE_(hOverE), ptError_(ptError) {
+      setPdgId(isEM ? 22 : 130);  // photon : non-photon(K0)
+    }
+
+    float hOverE() const { return hOverE_; }
```

Comment:
what about `const float`?

Reply:
You probably meant `const float &`.
Same as above, more efficient to return float type by value.

Reply:
Just to be clear, for value return types (i.e. `const float`) the `const` effectively has no meaning.